api/api/src/main/scala/uk/ac/wellcome/platform/api/models/DocumentFilter.scala

```diff
@@ -1,11 +1,34 @@
 package uk.ac.wellcome.platform.api.models
 
+import uk.ac.wellcome.models.work.internal.AccessStatus
+
 import java.time.LocalDate
 
 sealed trait DocumentFilter
 sealed trait WorkFilter extends DocumentFilter
 sealed trait ImageFilter extends DocumentFilter
 
+trait IncludeExcludeFilter[T] {
+  val include: List[T]
+  val exclude: List[T]
+
+  import com.sksamuel.elastic4s.ElasticDsl._
+
+  def toElasticString(obj: T): String
+
+  def toElasticQuery(field: String) =
+    (include.map(toElasticString(_)), exclude.map(toElasticString(_))) match {
+      case (include, Nil) =>
+        termsQuery(field = field, values = include)
+      case (Nil, exclude) =>
+        not(termsQuery(field = field, values = exclude))
+      case (include, exclude) =>
+        must(
+          termsQuery(field = field, values = include),
+          not(termsQuery(field = field, values = exclude)))
+    }
+}
```

Comment:
Generally we've kept all the Elastic stuff on the edge of where we interact with Elastic, leaving the responsibility of the conversion up to the elasticsearch code.

[`AggregationsRequest`](https://github.com/wellcomecollection/catalogue/blob/master/common/display/src/main/scala/uk/ac/wellcome/display/models/AggregationRequest.scala) and [`SortRequest`](https://github.com/wellcomecollection/catalogue/blob/master/common/display/src/main/scala/uk/ac/wellcome/display/models/SortRequest.scala) are examples of this.

Just a note that often I have found doing things in multiple ways make it hard to fathom which way to do it, or not do it, in a few months.

Reply:
Ok, will see what I can do